### DISCUSSION TITLE
Expose logging function within workflow for enhanced debugging

### DISCUSSION THREAD

**Discussion by gaberger** [mikub/titanoboa#25] [General]
It would be nice to log data from within a workflow to improve debugging

**Comment by mikub** [comment]
Yep, I was thinking about also capturing stdout/stderr of each function and adding it to the job history / showing it in the UI.
Currently if you want to see that you have to check the server log (which is accessible in UI as well), which is doable, but clumsy in clustered setup.

**Reply by gaberger** [reply]
Take a look at what the Argo project is doing here.

**Reply by mikub** [reply]
@gaberger would you have a link? I am going through some Argo demos etc. but still have not bumped into their logging, so maybe you know where to look.

Also thanks for pointing me to Argo, was not aware of it - it is always fun to find different takes on workflow automation, I like what Intuit/Argo did!
